framework/helpers/BaseStringHelper.php

```diff
@@ -104,12 +104,15 @@ public static function dirname($path)
      */
     public static function truncate($string, $length, $suffix = '...', $encoding = null, $asHtml = false)
     {
+        if ($encoding === null) {
+            $encoding = Yii::$app ? Yii::$app->charset : 'UTF-8';
```

Comment:
Even though it is happening often that still does not justify an extra method call. Besides degrading performance it makes the code depend on more things and harder to read.